stl/inc/xstring

```diff
@@ -3052,8 +3052,9 @@ private:
 
         if (_Right_data._Large_mode_engaged()) { // steal buffer
             _Construct_in_place(_My_data._Bx._Ptr, _Right_data._Bx._Ptr);
-            _Right_data._Bx._Ptr = nullptr;

```

Comment:
emm, for clarity it might be better just to put the correction `_Destroy_in_place(_Right_data._Bx._Ptr);` in this original place. apparently not worth resetting the testings though...